File: 991-broken-calculator/solution.js
//OJ--> https://leetcode.com/problems/broken-calculator/

var brokenCalc = function(start, target) {
    
    if(start >= target)
        return start-target;
    
    if(target%2 === 0)
        return 1 + brokenCalc(start, target/2);
       
    return 1 + brokenCalc(start, target+1);
    
};

File: 991-broken-calculator/solution.cpp
//OJ--> https://leetcode.com/problems/broken-calculator/

class Solution {
public:
        
    int brokenCalc(int start, int target) {
        
        
        
        if(target <= start){
            return start-target;
        }
        
        if(target %2 == 0)
            return 1 + brokenCalc(start, target/2);
        
        return 1+ brokenCalc(start, target+1);
        
    }
};